### ISSUE TITLE
FR: add setPriority

### ISSUE THREAD

**Issue by zonky2** [terminal42/contao-notification_center#183]
it would be nice if there was a way of setting/override the priority of email e.g.

```
$objNotification = NotificationCenter\Model\Notification::findByPk($intNotificationId);
if (null !== $objNotification) {
    $objNotification->setPriority($priority); // $priority = 1
    $objNotification->send($arrTokens, $strLanguage); // Language is optional
}
```
or

```
$objNotification = NotificationCenter\Model\Notification::findByPk($intNotificationId);
if (null !== $objNotification) {
    $objNotification->send($arrTokens, $strLanguage, $attachments, $priority);
}
```

**Comment by richardhj** [comment]
Priority of an email?

**Comment by zonky2** [comment]
yes
https://github.com/terminal42/contao-notification_center/blob/master/library/NotificationCenter/MessageDraft/EmailMessageDraft.php#L135-L147

**Comment by richardhj** [comment]
With the change introduced in #184, the following should work 🤔

```
foreach ($notification->getMessages() as $message) {
	if (null !== $gatewayModel = $message->getRelated('gateway') && $gatewayModel->getGateway() instanceof Email) {
	$message->email_priority = 1;
	}
}
```

**Comment by aschempp** [comment]
The notification does not have any priority, only the messages can have one, if they are emails. So editing the messages (`$message->email_priority`) seems correct to me.

**Comment by zonky2** [comment]
I`ll test it short time - thx

**Comment by zonky2** [comment]
Test with

```
            // Notification id 3.
            $notification = Notification::findByPk('3');
            if (null !== $notification) {
                foreach ($notification->getMessages() as $message) {
                    if (null !== $gatewayModel = $message->getRelated('gateway') && $gatewayModel->getGateway() instanceof Email) {
                        $message->email_priority = 1;
                    }
                }
                $notification->send($tokens);
            }
```
send error
```
Call to a member function getGateway() on null  
```

**Comment by zonky2** [comment]
fixed in C 4.8.3 https://github.com/contao/contao/commit/d19f747865194b6b101a986ca56ecaa369bcd82e
